Update ScratchNFT.sol

Mixelated opened the pull request.

Mixelated committed:
Update ScratchNFT.sol

vm06007 commented on contracts/ScratchNFT.sol:
```diff
@@ -92,8 +92,7 @@ abstract contract ScratchNFT is CommonNFT, PrizeTiers  {
                 "/",
                 claimDone,
-                "&edition=",
-                editionIdFromTicket.toString()
```
this change leads to unused local variable `editionIdFromTicket` and also do we need `editions` mapping in this case?

Some test needs to be adjusted

[FAIL. Reason: Assertion failed.] testTokenURI() (gas: 419156)
Logs:
  Error: a == b not satisfied [string]
        Left: 1/false.json
       Right: 1/false&edition=10

Mixelated replied:
yea we dont need edition anymore for purpose of tokenURI

Mixelated committed:
update test, remove editions

Mixelated committed:
add base uri extended

vm06007 commented on contracts/CommonNFT.sol:
```diff
@@ -21,6 +23,14 @@ abstract contract CommonNFT is ERC721Enumerable {
         _;
     }
 
+    function setBaseURI(string memory baseURI_) public onlyOwner {
+        _baseURIextended = baseURI_;
+    }
```
- it's better to mark this as `external` since this function has no internal use. (cleaner, cheaper)
- input paramter can be marked as `calldata` for a better use.
- it would be good to add tests for new functions.

vm06007 committed:
cleanup complications

vm06007 committed:
add updateBaseURI()

- allows to update baseURI

vm06007 committed:
🧪 add testUpdateBaseURI()

- so we can test new function updateBaseURI()
- so we can make sure onlyOwner can call it

vm06007 commented:
> _ No description provided. _

> good to add description
